### PR TITLE
Deprecate `.commit()`, `.rollback()` or `.close()` managed transactions

### PR DESCRIPTION
**Pull request by bigmontz**
Allowing self-manage managed transactions could causing unexpected behaviours in the user code depends on how the different possible code paths in the transaction work.

### CODE REVIEW

**Comment by bigmontz** [comment]
Depends on: https://github.com/neo4j-drivers/testkit/pull/415

**Comment by bigmontz** [comment]
> Depends on: [neo4j-drivers/testkit#415](https://github.com/neo4j-drivers/testkit/pull/415)

Not anymore

File: packages/core/src/transaction-managed.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  rollback(): Promise<void> {
+    return super.rollback()
+  }
+
+  /**
+   * Closes the transaction
+   *
+   * This method will roll back the transaction if it is not already committed or rolled back.
+   *
+   * @deprecated Close should not be done in transaction work. See {@link ManagedTransaction#commit} and {@link ManagedTransaction#rollback}
```
Comment (robsdedude) [inline packages/core/src/transaction-managed.ts]:
```suggestion
   * @deprecated Closure should not be done in transaction work. See {@link ManagedTransaction#commit} and {@link ManagedTransaction#rollback}
```

File: packages/neo4j-driver/src/transaction-managed-rx.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+   */
+  rollback () {
+    return super.rollback()
+  }
+
+  /**
+   * Closes the transaction
+   *
+   * This method will roll back the transaction if it is not already committed or rolled back.
+   * @deprecated Close should not be done in transaction work. See {@link RxManagedTransaction#commit} and {@link RxManagedTransaction#rollback}
```
Comment (robsdedude) [inline packages/neo4j-driver/src/transaction-managed-rx.js]:
```suggestion
   * @deprecated Closure should not be done in transaction work. See {@link RxManagedTransaction#commit} and {@link RxManagedTransaction#rollback}
```

File: packages/neo4j-driver/types/transaction-managed-rx.d.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+   * @deprecated Rollback should be done by throwing or returning a rejected promise from the transaction work.
+   * @returns {Observable} - An empty observable
+   */
+  rollback(): Observable<any>
+
+  /**
+   * Closes the transaction
+   *
+   * This method will roll back the transaction if it is not already committed or rolled back.
+   * @deprecated Close should not be done in transaction work. See {@link RxManagedTransaction#commit} and {@link RxManagedTransaction#rollback}
```
Comment (robsdedude) [inline packages/neo4j-driver/types/transaction-managed-rx.d.ts]:
```suggestion
   * @deprecated Closure should not be done in transaction work. See {@link RxManagedTransaction#commit} and {@link RxManagedTransaction#rollback}
```

**Review by robsdedude — APPROVED** [review]
:robot:

**Pull request merged**